src/build.c

```diff
@@ -1081,8 +1082,15 @@ static gchar *build_create_shellscript(const gchar *working_dir, const gchar *cm
 #ifdef G_OS_WIN32
 	/* Expand environment variables like %blah%. */
 	expanded_cmd = win32_expand_environment_variables(cmd);
-	str = g_strdup_printf("cd \"%s\"\n\n%s\n\n%s\ndel \"%%0\"\n\npause\n", working_dir, expanded_cmd, (autoclose) ? "" : "pause");
+	str = g_strdup_printf(
+		"cd \"%s\"\n\n\n%s\n\n%s\ndel \"%%0\"\n\npause\n",
+		working_dir, expanded_cmd, (autoclose) ? "" : "pause");
 	g_free(expanded_cmd);
+	/* Convert script content into system codepage */
+	utf8_script_content = win32_convert_to_system_codepage(str, error);
```

Comment:
> most of our problems appear to be that spawn_* uses windows calls and not glib calls and do not convert from the glib rule of always UTF8 to the real windows rule of windows locale

Yeah, AFAIK the "proper" way is to define `UNICODE` before including `windows.h` and then converting all of the GLib UTF-8 to Win32 UTF-16/wide char. ANSI code pages is a completely historical relic since around 1998 IIUC.